tests/integrationv2/test_signature_algorithms.py

```diff
@@ -176,7 +183,68 @@ def test_s2n_client_signature_algorithms(managed_process, cipher, provider, othe
         results.assert_success()
         assert to_bytes("Actual protocol version: {}".format(
             expected_version)) in results.stdout
-        assert signature_marker(
-            Provider.ServerMode, signature) in results.stdout or not server_sigalg_used
-        assert (signature_marker(Provider.ClientMode, signature)
-                in results.stdout) == client_auth
+        # TLS1.1 always negotiates to RSA+MD5_SHA1 signature algorithm.
+        if protocol < Protocols.TLS12:
+            assert signature_marker(
+                Provider.ServerMode, Signatures.MD5_SHA1
+            ) in results.stdout
+            assert (signature_marker(Provider.ClientMode, Signatures.MD5_SHA1)
+                    in results.stdout) == client_auth
+        else:
+            assert signature_marker(
+                Provider.ServerMode, signature) in results.stdout or not server_sigalg_used
+            assert (signature_marker(Provider.ClientMode, signature)
+                    in results.stdout) == client_auth
+
+
+@pytest.mark.uncollect_if(func=skip_ciphers)
+@pytest.mark.parametrize("cipher", [Ciphers.SECURITY_POLICY_20210816], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("protocol", [Protocols.TLS12], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+# RSA_SHA224 signature algorithm is not included in security_policy_20210816[].
+@pytest.mark.parametrize("signature", [Signatures.RSA_SHA224], ids=get_parameter_name)
```

Comment:
I'd rethink whether all your singleton parameters shouldn't be parameters. For example, this test will never be able to handle arbitrary combinations of cipher + signature. And will it ever apply to a protocol version other than TLS1.2?

It looks like your only parameters are provider (if we wanted to add another some day) and certificate. And certificate's not particularly interesting, since it already can only be RSA.

Reply:
In the updated test case, wouldn't it be good to them parametrized. It would be easier to make changes someday.

Reply:
But some of them can never change, right? Like the cipher and protocol version?

Reply:
After adding ECDSA case, signature is parameterized. This makes certificate parameterized too. I think it would be simpler/cleaner to keep as all of them parameterized.